File: mobile/src/main/java/rhedox/gesahuvertretungsplan/model/Boards.java
package rhedox.gesahuvertretungsplan.model;

import java.util.List;

/**
 * Created by robin on 28.09.2016.
 */

public class Boards {

	private List<Board> boards;

	public Boards(List<Board> boards) {
		this.boards = boards;
	}

	public List<Board> getBoards() {
		return boards;
	}

	public static class Board {
		private String name;

		public Board(String name) {
			this.name = name;
		}

		public String getName() {
			return name;
		}
	}
}

File: mobile/src/main/java/rhedox/gesahuvertretungsplan/model/database/tables/Boards.kt
package rhedox.gesahuvertretungsplan.model.database.tables

import android.database.sqlite.SQLiteDatabase

/**
 * Created by robin on 30.10.2016.
 */
sealed class Boards private constructor() {
    companion object {
    const val name = "boards";
    const val columnId = "id";
    const val columnName = "name"

    val availableColumns = setOf(
            columnId, columnName)

    fun onCreate(db: SQLiteDatabase) {
        val sql = """CREATE TABLE $name
            (
                $columnId INTEGER PRIMARY KEY AUTOINCREMENT,
                $columnName TEXT
            );
            """;
        db.execSQL(sql);
    }

    fun onUpgrade(db: SQLiteDatabase, oldVersion: Int, newVersion: Int) {
    }

    fun clear(db: SQLiteDatabase) {
        db.execSQL("DELETE FROM ${Boards.name} WHERE 1;");
    }
}
}